cloudsql/mysql/database-sql/connect_connector_iam_authn.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Google LLC
```

Comment:
Do we need a test for this sample?

Reply:
This is just a minor variation of https://github.com/GoogleCloudPlatform/golang-samples/blob/main/cloudsql/mysql/database-sql/connect_connector.go where the only difference is the addition of a `cloudsqlconn.WithIAMAuthN()` option. For the connector sample, we have a [test in place](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/cloudsql/mysql/database-sql/cloudsql_test.go#L164). So I think we're good actually.